### PR TITLE
fix(actor-core-typed): cap guardian startup deferred buffer

### PR DESCRIPTION
**Pull request by j5ik2o**
### Motivation
- `GuardianStartupActor` が起動完了前メッセージを無制限に `Vec` へ蓄えるため、guardian startup が進まないケースでメモリを消費し続ける可能性があった。
- 起動前 deferral の失敗後に restart した場合、inner actor を起動済み扱いにしてしまうと、その後の `GuardianStartupStart` で inner startup が正しく実行されない。

### Description
- 起動前 deferred buffer に 1024 件の上限を追加。
- 上限到達時と active current message がない deferral 不変条件違反は、supervision policy が扱える `Recoverable` な typed error として返す。
- startup deferral 由来の restart では `started = false` を維持し、deferred queue を破棄して `GuardianStartupStart` まで inner actor の `pre_start` / `post_restart` を呼ばないようにした。
- overflow、active current message 不在、deferral restart 後の startup gate 維持を回帰テストで固定。

### Verification
- `MISE_TRUSTED_CONFIG_PATHS=$PWD/mise.toml cargo fmt --all --check`
- `MISE_TRUSTED_CONFIG_PATHS=$PWD/mise.toml cargo test -p fraktor-actor-core-typed-rs guardian_startup`
- `MISE_TRUSTED_CONFIG_PATHS=$PWD/mise.toml ./scripts/ci-check.sh ai all`

---

> [!NOTE]
> **Medium Risk**
> Changes startup gating and restart behavior in `GuardianStartupActor`, which can affect message delivery ordering and supervision/restart semantics if edge cases are missed. Scope is contained and covered by new unit tests.
> 
> **Overview**
> **Typed guardian startup deferral is now bounded and restart-safe.** `GuardianStartupActor` caps the deferred pre-start message buffer at `1024` and returns *typed recoverable* errors when deferral invariants are violated (no active current message) or the buffer overflows.
> 
> On `post_restart`, deferral-originated errors now keep `started = false` and drop the deferred queue so the inner actor’s startup runs only after a subsequent `GuardianStartupStart`. Adds regression tests for missing active message, overflow behavior, and deferral-restart startup gating.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8fe63a3f49f2ad8d311f1708d591c2039f9386dc. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: modules/actor-core-typed/src/internal/guardian_startup_actor.rs [hunk 1]
```diff
@@ -78,6 +82,9 @@ impl GuardianStartupActor {
     let Some(message) = ctx.clone_current_message() else {
       return Err(ActorError::recoverable("guardian startup deferral requires an active user message"));
     };
+    if self.deferred.len() >= GUARDIAN_STARTUP_DEFERRED_LIMIT {
+      return Err(ActorError::recoverable(GUARDIAN_STARTUP_DEFERRED_OVERFLOW_REASON));
+    }
```
[the remark this answers is not in the document]
Comment (j5ik2o) [reply modules/actor-core-typed/src/internal/guardian_startup_actor.rs:99]: 対応しました。起動前の deferred buffer overflow は recoverable restart ではなく fatal error として返すように変更し、GuardianStartupStart 前の restart によって deferred messages や inner.pre_start() が飛ぶ経路をなくしました。あわせて overflow が Fatal になる回帰テストを追加しています。
